frappe_telegram/frappe_telegram/doctype/telegram_bot/test_telegram_bot.py

```diff
@@ -1,8 +1,137 @@
 # Copyright (c) 2021, Leam Technology Systems and Contributors
 # See license.txt
 
-# import frappe
 import unittest
 
+import frappe
+from frappe_telegram.frappe_telegram.doctype.telegram_bot import DEFAULT_TELEGRAM_BOT_KEY
+from frappe_telegram.utils.test_fixture import TestFixture
+
+
+class TelegramBotFixtures(TestFixture):
+    def __init__(self):
+        super().__init__()
+        self.DEFAULT_DOCTYPE = "Customer Document Type"
+
+    def make_fixtures(self):
+
+        # Some fixtures that have not been inserted
+
+        fix1 = frappe.get_doc(frappe._dict(
+            doctype="Telegram Bot",
+            title="TestBotFix1",
+            api_token="randonapitoken"
+        ))
+        self.add_document(fix1)
+
+        fix2 = frappe.get_doc(frappe._dict(
+            doctype="Telegram Bot",
+            title="TestBotFix2",
+            api_token="otherrandonapitoken"
+        ))
+        self.add_document(fix2)
+
+        fix3 = frappe.get_doc(frappe._dict(
+            doctype="Telegram Bot",
+            title="TestBotFix3",
+            api_token="anotherrandonapitoken"
+        ))
+        self.add_document(fix3)
+
+
 class TestTelegramBot(unittest.TestCase):
-	pass
+
+    telegram_bots = TelegramBotFixtures()
+
+    def setUp(self):
+        self.telegram_bots.setUp()
+
+    def tearDown(self):
+        self.telegram_bots.tearDown()
+
+    def test_auto_default_on_first_bot(self):
+        """
+        Test that if a new bot is created, it is marked as default if no other bot exists
+        Also test that if that default is deleted, another one is marked default
+        """
+
+        existing_bots = frappe.get_all("Telegram Bot")
+        self.assertEqual(len(existing_bots), 0)  # Test assumes there are no bots existing
```

Comment:
This is really harsh 😅 
I am so lazy to re-obtain the tokens & re-create my bots 😭 
Is this assumption really required ?

Reply:
Unfortunately a necessity for this test, as we need to test what happens when the first one is created. Unless...
I back up the existing bots first and restore them after (PR for frappe_testing incoming 😄)

Reply:
hehe ok